VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -0,0 +1,246 @@
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.RequestManager;
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.model.api.ProfileModel;
+import org.edx.mobile.module.prefs.PrefManager;
+import org.edx.mobile.third_party.iconify.IconDrawable;
+import org.edx.mobile.third_party.iconify.Iconify;
+import org.edx.mobile.user.Account;
+import org.edx.mobile.user.GetAccountTask;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.util.Locale;
+
+import roboguice.fragment.RoboFragment;
+import roboguice.inject.InjectExtra;
+
+public class UserProfileFragment extends RoboFragment {
+
+    @InjectExtra(UserProfileActivity.EXTRA_USERNAME)
+    private String username;
+
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    @Nullable
+    private Account account;
+
+    @Nullable
+    private ViewHolder viewHolder;
+
+    @Nullable
+    MenuItem editProfileMenuItem;
+
+    @Inject
+    private Router router;
+
+    private boolean isViewingOwnProfile;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+        setHasOptionsMenu(true);
+
+        final ProfileModel model = new PrefManager(getActivity(), PrefManager.Pref.LOGIN).getCurrentUserProfile();
+        isViewingOwnProfile = null != model && model.username.equalsIgnoreCase(username);
+
+        getAccountTask = new GetAccountTask(getActivity(), username) {
+            @Override
+            protected void onSuccess(Account account) throws Exception {
+                UserProfileFragment.this.account = account;
+                if (null != viewHolder) {
+                    viewHolder.setAccount(account);
+                }
+            }
+        };
+        getAccountTask.setTaskProcessCallback(null); // Disable default loading indicator, we have our own
+        getAccountTask.execute();
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        super.onCreateOptionsMenu(menu, inflater);
+        if (isViewingOwnProfile) {
+            inflater.inflate(R.menu.edit_profile, menu);
+            editProfileMenuItem = menu.findItem(R.id.edit_profile).setIcon(
+                    new IconDrawable(getActivity(), Iconify.IconValue.fa_pencil)
+                            .actionBarSize().colorRes(R.color.edx_white));
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.edit_profile) {
```

Comment:
This should probably be a switch statement, in case we add more menu items later.